### PR TITLE
feat: increase coverage for contracts package

### PR DESCRIPTION
**Pull request by hweawer**
# Summary
                                                                                                                                         
  - Comprehensive unit test coverage for the contracts package (tests/execution/contracts/test_contract_transformations.py, ~1 550 ines): covers data transformations, ABI decoding, batching logic, and edge cases for every contract wrapper (DepositContract, LidoContract, BurnerContract, CuratedStakingModuleContract, MetaRegistryContract, OracleReportSanityCheckerContract, WithdrawalQueueNftContract, DelegationContract, and others).                                                                           
  - Integration test suite expanded: new or extended test files for BaseOracleContract, BurnerContract, CuratedStakingModuleContract, DepositContract, HashConsensusContract, MetaRegistryContract, VaultHubContract, and integration conftest.py fixtures.                  
  - Bug fixes surfaced by tests:                                                                                     
    - StakingRouterContract.get_staking_module — new method wrapping getStakingModule(id).                                               
    - VaultHubContract event methods — added logger.info('Call …') so check_contract's log-count assertion passes.                       
    - make_checker in contract_utils — unwraps NewType before calling isinstance() (matches existing logic in check_dataclass_types).    
    - caplog.clear() added before check_contract in tests that pre-call a contract method to obtain fixture data (test_hash_consensus, test_base_oracle, test_meta_registry).

### CODE REVIEW

File: src/providers/execution/contracts/staking_router.py [hunk 1]
```diff
@@ -67,6 +68,22 @@ def get_all_node_operator_digests(
...
+        logger.info(
+            {
+                'msg': f'Call `getStakingModule({staking_module_id})`.',
+                'value': response,
+                'block_identifier': repr(block_identifier),
+                'to': self.address,
+            }
+        )
+        return named_tuple_to_dataclass(response, StakingModule.from_response)
+
```
[the remark this answers is not in the document]
Comment (hweawer) [reply src/providers/execution/contracts/staking_router.py]: @copilot apply changes based on [this feedback](https://github.com/lidofinance/lido-oracle/pull/890#discussion_r3064846269)

**Review by F4ever — APPROVED** [review]
